gulpfile.js

```diff
@@ -151,23 +151,49 @@ gulp.task('create-7zip-archive', function(cb) {
   exec(packCmd, createExecCallback(cb, true));
 });
 
-gulp.task('update-devstudio-version', function(cb) {
-  let url = reqs['jbds.jar'].url.substring(0, reqs['jbds.jar'].url.lastIndexOf("/")) + '/content.json';
-  request(url, function(err, response, body) {
-    if (err) {
-      cb(err);
-    } else {
-      let versionRegex = /(\d+\.\d+\.\d+\.\w+\d*).*/;
-      let finalVersion = versionRegex.exec(body)[1];
+gulp.task('update-requirements',['create-modules-link'], function() {
 
-      if (reqs['jbds.jar'].version != finalVersion) {
-        reqs['jbds.jar'].version = finalVersion;
-        fs.writeFile('./requirements.json', JSON.stringify(reqs, null, 2), cb);
-      } else {
-        cb();
+  let updateDevStudioVersion = ()=>{
+    return new Promise((resolve,reject) => {
+      let url = reqs['jbds.jar'].url.substring(0, reqs['jbds.jar'].url.lastIndexOf("/")) + '/content.json';
+      request(url, (err, response, body)=>{
+        if (err) {
+          reject(err);
+        } else {
+          let versionRegex = /(\d+\.\d+\.\d+\.\w+\d*).*/;
+          let finalVersion = versionRegex.exec(body)[1];
+
+          if (reqs['jbds.jar'].version != finalVersion) {
+            reqs['jbds.jar'].version = finalVersion;
+          }
+          resolve()
+        }
+      });
+    });
+  };
+
+  let updateDevStudioSha = ()=>{
+    return new Promise((resolve,reject) => {
+      let url = reqs['jbds.jar'].sha256sum;
+      if (url.length == 64 && url.indexOf("http")<0 && url.indexOf("ftp")<0) {
+        resolve();
+      } else {
+        request(url, (err, response, body) => {
+          reqs['jbds.jar'].sha256sum = body;
+          resolve();
+        });
       }
-    }
-  });
+    });
+  };
+
+  return Promise.resolve()

```

Comment:
Well, if you look at the model classes, the promise chains usually start with the first business function. I'm not a fan of this padded construction (but it's no big deal either).